url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/DemandeMoyen.java

```diff
@@ -0,0 +1,152 @@
+package org.agetac.common.model.impl;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.impl.Vehicule.CategorieVehicule;
+import org.agetac.common.model.sign.AbstractModel;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class DemandeMoyen extends AbstractModel {
+
+	public enum EtatDemande {
+	 LANCEE, REFUSEE, ACCEPTEE
+	}
+	
+	private static final List<CategorieVehicule> VALUES_CategorieVehicule = Collections.unmodifiableList(Arrays.asList(CategorieVehicule.values()));
+	private static final int SIZE_CategorieVehicule = VALUES_CategorieVehicule.size();
+	
+	private static final List<EtatDemande> VALUES_EtatDemande = Collections.unmodifiableList(Arrays.asList(EtatDemande.values()));
+	private static final int SIZE_EtatDemande = VALUES_EtatDemande.size();
+	
+	private EtatDemande etat;
+	private String groupeID;
+	private HashMap<EtatDemande, String> groupesHoraires; 
+	private CategorieVehicule categorie;
+	
+	private String vehId = null; // l'id du vehicule rďż˝sultat de la demande
+	
+	public String getVehId() {
+		return vehId;
+	}
+
+	public void setVehId(String vehId) {
+		this.vehId = vehId;
+	}
+
+	public DemandeMoyen() {
+		super(null,"demande",new Position(0,0));
+		this.etat = EtatDemande.LANCEE;
+		this.groupeID = "0";
+		this.groupesHoraires = new HashMap<EtatDemande, String>();
+		this.groupesHoraires.put(EtatDemande.LANCEE, "");
+		this.groupesHoraires.put(EtatDemande.REFUSEE, "");
+		this.groupesHoraires.put(EtatDemande.ACCEPTEE, "");
+		this.categorie = null;
+		this.vehId = null;
+	}
+	
+	public DemandeMoyen(String uid, Position position, CategorieVehicule cat, EtatDemande etat, Groupe groupe, String heure) {
+		super(uid, cat.name()+"-demande", position);
+		this.categorie = cat;
+		this.etat = etat;
+		this.groupeID = groupe.getUniqueID();
+		this.groupesHoraires = new HashMap<EtatDemande, String>();
+		this.groupesHoraires.put(EtatDemande.LANCEE, heure);
+		this.groupesHoraires.put(EtatDemande.REFUSEE, "");
+		this.groupesHoraires.put(EtatDemande.ACCEPTEE, "");
+		this.vehId = null;
+	}
+	
+
+	
+	public DemandeMoyen(String uid, String nom, Position position, EtatDemande etat, String groupeID) {
+		super(uid, nom, position);
+		this.etat = etat;
+		this.groupeID = groupeID;
+		this.groupesHoraires = new HashMap<EtatDemande, String>();
+		this.vehId = null;
+	}
+	
+	public DemandeMoyen(JSONObject json) throws InvalidJSONException {
+		super(json);
+		try{
+			this.categorie = CategorieVehicule.valueOf(json.getString("categorie"));
+			this.etat = EtatDemande.valueOf(json.getString("etat"));
+			this.groupeID = json.getString("groupeID");
+			//TODO: ComplĂŠter avec la liste des groupe horaire
+			//this.vehId = json.getString("vehId");
+			this.groupesHoraires = new HashMap<EtatDemande, String>();
+		}catch(JSONException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)